statsd/src/main/java/com/avast/metrics/statsd/StatsDMetricsMonitorBuilder.java

```diff
@@ -0,0 +1,79 @@
+package com.avast.metrics.statsd;
+
+import com.avast.metrics.api.Naming;
+import com.avast.metrics.filter.MetricsFilter;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+public class StatsDMetricsMonitorBuilder {
+		private String host;
+		private int port;
+		private boolean autoRegisterMetrics = false;
+		private String prefix;
+		private Naming naming = Naming.defaultNaming();
+		private Duration gaugeSendPeriod = getDefaultGaugeSendPeriod();
+		private ScheduledExecutorService scheduler = createScheduler();
+		private MetricsFilter metricsFilter = MetricsFilter.ALL_ENABLED;
+
+		public StatsDMetricsMonitorBuilder setHost(String host) {
+				this.host = host;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setPort(int port) {
+				this.port = port;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setAutoRegisterMetrics(boolean autoRegisterMetrics) {
+				this.autoRegisterMetrics = autoRegisterMetrics;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setPrefix(String prefix) {
+				this.prefix = prefix;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setNaming(Naming naming) {
+				this.naming = naming;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setGaugeSendPeriod(Duration gaugeSendPeriod) {
+				this.gaugeSendPeriod = gaugeSendPeriod;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setScheduler(ScheduledExecutorService scheduler) {
+				this.scheduler = scheduler;
+				return this;
+		}
+
+		public StatsDMetricsMonitorBuilder setMetricsFilter(MetricsFilter metricsFilter) {
+				this.metricsFilter = metricsFilter;
+				return this;
+		}
+
+		public StatsDMetricsMonitor createStatsDMetricsMonitor() {
+				return new StatsDMetricsMonitor(host, port, autoRegisterMetrics, prefix, naming, gaugeSendPeriod, scheduler, metricsFilter);
+		}
+
+		public StatsDMetricsMonitor createStatsDMetricsMonitor(StatsDMetricsMonitor monitor, String... newNames) {
```

Comment:
What is purpose of this method? I would expect following of the builder pattern and a method `withMonitor()`. Tree of names are created by calling `named()`, it is out of scope of the builder, it's API of the instance.